packages/datx-jsonapi/src/decorateCollection.ts

```diff
@@ -175,6 +173,33 @@ export function decorateCollection(
       return read<T>(query.url, this, reqOptions).then(handleErrors);
     }
 
+    public async getAll<T extends IJsonapiModel = IJsonapiModel>(
+      type: IType | IModelConstructor<T>,
+      options?: IRequestOptions,
```

Comment:
Should we add some safeguards for infinite loops?
```
maxRequests: 500
```

While loops are dangerous because sometimes caching on the server can cause that `next` page has the same link as the current page. 
We can prevent deadlocks with some guards.

Another example is that page count could increase over time.
In the beginning, we could have only a few pages, but the user could enter more and more stuff in the CMS, and request time could rise sky-high.